op-service/sources/reth_db.go

```diff
@@ -68,26 +71,40 @@ func FetchRethReceipts(dbPath string, blockHash *common.Hash) (types.Receipts, e
 	return receipts, nil
 }
 
+func OpenDBReadOnly(dbPath string) (db unsafe.Pointer, err error) {
+	// Convert the db path to a C string and defer its deallocation
+	cDbPath := C.CString(dbPath)
+	defer C.free(unsafe.Pointer(cDbPath))
+
+	// Call the C function to fetch the receipts from the Reth Database
+	openDBResult := C.open_db_read_only(cDbPath)
+
+	if openDBResult.error {
+		return nil, fmt.Errorf("Error open db from Reth Database.")
+	}
+
+	return openDBResult.data, nil
+}
+
 type RethDBReceiptsFetcher struct {
-	dbPath string
-	// TODO(8225): Now that we have reading from a Reth DB encapsulated here,
-	//   We could store a reference to the RethDB here instead of just a db path,
-	//   which would be more optimal.
-	//   We could move the opening of the RethDB and creation of the db reference
-	//   into NewRethDBReceiptsFetcher.
+	dbInstance unsafe.Pointer
 }
 
 var _ ReceiptsProvider = (*RethDBReceiptsFetcher)(nil)
 
 func NewRethDBReceiptsFetcher(dbPath string) *RethDBReceiptsFetcher {
+	db, _ := OpenDBReadOnly(dbPath)
```

Comment:
Could we make this function faillible (i.e. it returns `(*RethDBReceiptsFetcher, error) ` and report the error earlier?

Reply:
if we do this, we need also make `newRecProviderFromConfig` return error, and check error in `NewEthClient` in `eth_client.go` , and always return error with nil in `receipts_rpc.go`.It seems feasible

> Could we make this function faillible (i.e. it returns `(*RethDBReceiptsFetcher, error) ` and report the error earlier?

Reply:
I use a more simple way to implement this. Return nil when error and check nil in the `NewEthClient` function

> Could we make this function faillible (i.e. it returns `(*RethDBReceiptsFetcher, error) ` and report the error earlier?